14065 Remove whitespace characters from internationalized Dice Summary strings

rlament opened the pull request:
Removed the linefeed characters from the Dice Summary delimiters in the i18n file. Without digging into command parsing, it looks like the linefeed characters confused the parser. This then blocks the update of the die roll result on the other player's screen.

Leading and trailing whitespace in i18n properties files is problematic. These strings can be modified inadvertently. Some editors can automatically remove trailing plain whitespace depending on their configuration. Whitespace can be escaped but some editors may interpret and replace these. Leading or trailing whitespace may not be obvious to translators. Best to avoid leading and trailing whitespace in the i18n files.

As an aside, the adoc should indicate that the summary is internationalized and parsing should be discouraged.

riverwanderer commented:
Thank you @rlament! I've no idea how linefeed characters got in there (at least to my recollection) but I have come across trailing space characters. I will avoid them myself in future.

rlament commented:
I modified the DiceButton documentation to note that the summary is localized text and fixed some typos. Added a note to the effect that the name property is the Description field. I'll leave it to a reviewer to decide whether or not this closes #13980.

rlament commented:
> There's an issue with the test builds...

Well, this is the first time I've tried to use a "test build". I'm seeing the same issue, a pop-up saying "The application requires a Java Runtime Environment". I'm focusing on just the Windows x86_64 builds. So I went back in time and tried the older 3.7.16 builds. I can't get any of them to run; same error message. Then I tried the last 3.8.0 build; also a no go. In desperation, copied the jre/bin and jre/lib folders from 3.7.16 to the build folder. That launches the module manager. But then trying to open or edit a module produces the "Possible Anti-Virus Program Interference" pop-up. At this point not sure why none of the builds run. I don't have a working baseline to test against. With my lack of experience I don't know where things have gone astray.

riverwanderer commented:
When I am working on Vassal it is usually through IntelliJ and pushing against the 3.7.x build generally works for me. I am not in a position to try replicate your builds myself at the moment. Perhaps @uckelman can  confirm that the build target for this PR was correct and/or what might have gone awry with including Java.

rlament commented:
> Original reporter confirmed that this PR fixes #[14065](https://github.com/vassalengine/vassal/issues/14065). Solution to #[13980](https://github.com/vassalengine/vassal/issues/13980) also accepted.

@riverwanderer Did you also confirm that the summary text is still in the same sort order and format as before? I did test with a custom module and it did look ok to me.

uckelman commented on vassal-app/src/main/java/VASSAL/build/module/DiceButton.java:
```diff
@@ -50,10 +50,9 @@
-import java.util.Map;
-
+import java.util.SortedMap;
+import java.util.TreeMap;
```
What is the purpose of the changes to `DiceButton`?

riverwanderer replied:
The overall purpose (released with 3.7.16) was to provide as "summary" property that contains the roll results in the format requested in #[4427](https://github.com/vassalengine/vassal/issues/4427).

uckelman replied:
Yes, but what do the changes to `DiceButton` do?

uckelman commented on vassal-app/src/main/java/VASSAL/build/module/RandomTextButton.java:
```diff
@@ -164,8 +164,12 @@ protected void DR() {
+      }
       summaryResult.append(summaryKeys[i])
+              .append(' ')
               .append(Resources.getString("Dice.summary_times"))
```
What is the purpose of the changes to `RandomTextButton`?

riverwanderer replied:
As with DiceButton, the overall purpose (released with 3.7.16) was to provide as "summary" property that contains the roll results in the format requested in #[4427](https://github.com/vassalengine/vassal/issues/4427).

uckelman replied:
That doesn't answer my question. What I want to know is what these particular changes are for. What would happen were they not here at all?
